lib/benchee/utility/task.ex

```diff
@@ -0,0 +1,49 @@
+defmodule Benchee.Utility.Task do
+  @moduledoc """
+  So far this is the best way we have of controlling the effects of garbage
+  collection. We essentially re-implement the behavior of `Task.async/1` and
+  `Task.await/2` with our own spawning and monitoring of processes so we can use
+  Erlang's `erlang.spawn_opt/2` to control the garbage collection settings.
+  """
+
+  @doc """
+  Start a process that executes a function and  sends the result back to the
+  calling process
+  """
+  @spec async(fun) :: {pid, reference}
+  def async(fun) do
+    me = self()
+
+    Process.spawn(fn -> send(me, {self(), fun.()}) end, [
+      :monitor,
+      {:fullsweep_after, 999_999_999_999},
+      {:min_heap_size, 999_999_999_999}
```

Comment:
:D Yeah we definitely need to make this value configurable and we also definitely need to make this optional when the feature lands (cause integration tests etc.)